core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -955,6 +977,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * The variable will be sent to each cluster only once.
    */
   def broadcast[T: ClassTag](value: T): Broadcast[T] = {
+    assertNotStopped()
+    if (classOf[RDD[_]].isAssignableFrom(classTag[T].runtimeClass)) {
```

Comment:
I've changed this in my latest patch; we log a warning here and any errors are caught by the more general "display an error about RDD nesting if the `sc` field is null" check.